### PR TITLE
fix(babel): instrument istanbul coverage with babel

### PR DESCRIPTION
**Pull request by trevtrich**
- this fixes the coverage being reported in babelified code, and only
instruments in teh test environment
- this should be safe to add prior to [this pull
request](https://github.com/form8ion/javascript-core/pull/114) since the test
env just wont be set
- fixes form8ion/javascript-core#84

### CODE REVIEW

**Comment by trevtrich** [comment]
It feels a little awkward to force you to tie in `istanbul` if you use `babel` at all in your JS project, but I'm not sure we have a great way (or appetite, for that matter) to avoid this as it stands right now.

**Comment by travi** [comment]
>  if you use babel at all in your JS project

we should also be able to determine if the project will be unit tested. maybe we only include if that is true?

File: src/config/babel.js [hunk 1]
```diff
@@ -9,7 +9,10 @@ export default async function ({projectRoot, preset, transpileLint}) {
   }
 
   if (preset) {
-    await fsPromises.writeFile(`${projectRoot}/.babelrc`, JSON.stringify({presets: [preset.name], ignore: ['./lib/']}));
+    await fsPromises.writeFile(
+      `${projectRoot}/.babelrc`,
+      JSON.stringify({presets: [preset.name], ignore: ['./lib/'], env: {test: {plugins: ['istanbul']}}})
```
Comment (travi) [inline src/config/babel.js:14]: does some sort of dependency need to be installed to support the `istanbul` plugin here?

**Comment by trevtrich** [comment]
I like both of those thoughts. It at least makes more sense to assume if you are unit testing that you are using nyc, since we have the built in at the `unitTestScaffolder`. 

As for the dependency, right now it's inside `javascript-core` as part of the `nyc` scaffolder, but I think I lean with you, that this is probably better suited since it's where it's actually required. I'll get those swapped.

**Review by travi — APPROVED** [review]

File: src/config/babel.js [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
   }
 
   if (preset) {
-    await fsPromises.writeFile(`${projectRoot}/.babelrc`, JSON.stringify({presets: [preset.name], ignore: ['./lib/']}));
+    await fsPromises.writeFile(
+      `${projectRoot}/.babelrc`,
+      JSON.stringify({
+        presets: [preset.name],
+        ignore: ['./lib/'],
+        ...tests.unit && {env: {test: {plugins: ['istanbul']}}}
```
Comment (trevtrich) [inline src/config/babel.js:17]: This here is the new piece.

File: src/config/babel.js [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
+      })
+    );
   } else warn('Not configuring transpilation since no babel-config was provided');
 
   return {
-    devDependencies: ['@babel/register', ...preset ? [preset.packageName] : []],
+    devDependencies: [
+      '@babel/register',
```
Comment (travi) [inline src/config/babel.js:24]: i know this was before your change, but it seems like `@babel/register` should consider the `preset` presence too. am i crazy thinking that?

File: src/config/babel.js (hunk 2 above)
Comment (travi) [reply src/config/babel.js:17]: yep, makes sense

File: src/config/babel.js (hunk 3 above)
Comment (trevtrich) [reply src/config/babel.js:24]: i could definitely see that making some sense. actually it looks like if we did go that route, that could be part of the top level early-out code so we could clean up the rest of it as well

Comment (travi) [reply src/config/babel.js:24]: yeah, cool. some of these have gotten more complex over time, so some fresh eyes to clean up situations like that are always helpful.

totally cool with that cleanup being separate from this, but i'd like us not to lose track of it if possible

**Pull request merged**